### ISSUE TITLE
Stuck At Painting

### ISSUE THREAD

**Issue by JosiMou** [lllyasviel/style2paints#131]
Everytime I try to conduct the painting process, it gets stuck at painting, without yielding any results.

Anyone experienced the same?

![Annotation 2019-10-22 213300](https://user-images.githubusercontent.com/48890750/67285994-8d944900-f513-11e9-8f87-05e361af00e5.png)

**Comment by viet1pro** [comment]
Can you upload your log?

**Comment by JosiMou** [comment]
> Can you upload your log?

(animne) PS C:\Users\Joe\Desktop\style2paints\V3\server> python server.py                                               2019-10-22 22:56:00.812826: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
Using TensorFlow backend.
2019-10-22 22:56:02.239308: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library nvcuda.dll
2019-10-22 22:56:03.159699: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties:
name: GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.56
pciBusID: 0000:01:00.0
2019-10-22 22:56:03.164176: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
2019-10-22 22:56:03.168171: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_100.dll
2019-10-22 22:56:03.173926: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_100.dll
2019-10-22 22:56:03.177917: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_100.dll
2019-10-22 22:56:03.182424: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_100.dll
2019-10-22 22:56:03.189146: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_100.dll
2019-10-22 22:56:03.203039: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2019-10-22 22:56:03.206146: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2019-10-22 22:56:03.207599: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2
2019-10-22 22:56:03.212957: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties:
name: GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.56
pciBusID: 0000:01:00.0
2019-10-22 22:56:03.216076: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
2019-10-22 22:56:03.218399: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_100.dll
2019-10-22 22:56:03.220969: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_100.dll
2019-10-22 22:56:03.224638: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_100.dll
2019-10-22 22:56:03.228370: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_100.dll
2019-10-22 22:56:03.231937: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_100.dll
2019-10-22 22:56:03.235996: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2019-10-22 22:56:03.255798: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2019-10-22 22:56:03.688633: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-10-22 22:56:03.691871: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0
2019-10-22 22:56:03.695601: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N
2019-10-22 22:56:03.698676: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 2921 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)
WARNING:tensorflow:From C:\Users\Joe\Desktop\style2paints\V3\server\ai.py:44: The name tf.placeholder is deprecated. Please use tf.compat.v1.placeholder instead.

WARNING:tensorflow:From C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\tensorflow_core\python\ops\resource_variable_ops.py:1630: calling BaseResourceVariable.__init__ (from tensorflow.python.ops.resource_variable_ops) with constraint is deprecated and will be removed in a future version.
Instructions for updating:
If using Keras pass *_constraint arguments to layers.
C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\keras\engine\saving.py:341: UserWarning: No training configuration found in save file: the model was *not* compiled. Compile it manually.
  warnings.warn('No training configuration found in save file: '
WARNING:tensorflow:From C:\Users\Joe\Desktop\style2paints\V3\server\ai.py:53: where (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where
WARNING:tensorflow:From C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\keras\backend\tensorflow_backend.py:4070: The name tf.nn.max_pool is deprecated. Please use tf.nn.max_pool2d instead.

WARNING:tensorflow:From C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\keras\backend\tensorflow_backend.py:4074: The name tf.nn.avg_pool is deprecated. Please use tf.nn.avg_pool2d instead.

WARNING:tensorflow:From C:\Users\Joe\Desktop\style2paints\V3\server\ai.py:74: The name tf.image.resize_bilinear is deprecated. Please use tf.compat.v1.image.resize_bilinear instead.

WARNING:tensorflow:From C:\Users\Joe\Desktop\style2paints\V3\server\ai.py:89: The name tf.global_variables_initializer is deprecated. Please use tf.compat.v1.global_variables_initializer instead.

Bottle v0.12.17 server starting up (using PasteServer())...
Listening on http://0.0.0.0:80/
Hit Ctrl-C to quit.

serving on 0.0.0.0:80 view at http://127.0.0.1:80
127.0.0.1 - - [22/Oct/2019:22:56:36 +0900] "POST /get_sample_list HTTP/1.1" 200 1144 "http://127.0.0.1/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36"
127.0.0.1 - - [22/Oct/2019:22:56:43 +0900] "GET /samples/Apr19H22M59S23R132/options.sample.json HTTP/1.1" 304 - "http://127.0.0.1/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36"
127.0.0.1 - - [22/Oct/2019:22:56:43 +0900] "GET /res/raw-assets/texture/ring.png?t=0.10058645478013051 HTTP/1.1" 200 113227 "http://127.0.0.1/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36"
127.0.0.1 - - [22/Oct/2019:22:56:43 +0900] "GET /samples/Apr19H22M59S23R132/sketch.colorization.jpg HTTP/1.1" 304 - "http://127.0.0.1/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36"
127.0.0.1 - - [22/Oct/2019:22:56:43 +0900] "GET /samples/Apr19H22M59S23R132/composition.sample.jpg HTTP/1.1" 304 - "http://127.0.0.1/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36"
127.0.0.1 - - [22/Oct/2019:22:56:43 +0900] "GET /samples/Apr19H22M59S23R132/result.sample.jpg HTTP/1.1" 304 - "http://127.0.0.1/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36"
copy game/samples/Apr19H22M59S23R132 to game/rooms/Oct22H22M56S54R813
request result room = Oct22H22M56S54R813, ID = H22M56S54
processing painting in game/rooms/Oct22H22M56S54R813
sketch prepared
2019-10-22 22:56:55.518106: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2019-10-22 22:56:56.590973: E tensorflow/stream_executor/cuda/cuda_dnn.cc:329] Could not create cudnn handle: CUDNN_STATUS_ALLOC_FAILED
2019-10-22 22:56:56.598614: E tensorflow/stream_executor/cuda/cuda_dnn.cc:329] Could not create cudnn handle: CUDNN_STATUS_ALLOC_FAILED
2 root error(s) found.
  (0) Unknown: Failed to get convolution algorithm. This is probably because cuDNN failed to initialize, so try looking to see if a warning log message was printed above.
         [[node model_1/c0/convolution (defined at C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\tensorflow_core\python\framework\ops.py:1748) ]]
  (1) Unknown: Failed to get convolution algorithm. This is probably because cuDNN failed to initialize, so try looking to see if a warning log message was printed above.
         [[node model_1/c0/convolution (defined at C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\tensorflow_core\python\framework\ops.py:1748) ]]
         [[concat_3/_5195]]
0 successful operations.
0 derived errors ignored.

Original stack trace for 'model_1/c0/convolution':
  File "server.py", line 16, in <module>
    from ai import *
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\Users\Joe\Desktop\style2paints\V3\server\ai.py", line 55, in <module>
    baby_op = YUV2RGB(baby(tf.concat([ip1, baby_hint], axis=3)))
  File "C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\keras\engine\base_layer.py", line 489, in __call__
    output = self.call(inputs, **kwargs)
  File "C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\keras\engine\network.py", line 583, in call
    output_tensors, _, _ = self.run_internal_graph(inputs, masks)
  File "C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\keras\engine\network.py", line 740, in run_internal_graph
    layer.call(computed_tensor, **kwargs))
  File "C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\keras\layers\convolutional.py", line 171, in call
    dilation_rate=self.dilation_rate)
  File "C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\keras\backend\tensorflow_backend.py", line 3717, in conv2d
    **kwargs)
  File "C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\tensorflow_core\python\ops\nn_ops.py", line 898, in convolution
    name=name)
  File "C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\tensorflow_core\python\ops\nn_ops.py", line 1009, in convolution_internal
    name=name)
  File "C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\tensorflow_core\python\ops\gen_nn_ops.py", line 1071, in conv2d
    data_format=data_format, dilations=dilations, name=name)
  File "C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\tensorflow_core\python\framework\op_def_library.py", line 794, in _apply_op_helper
    op_def=op_def)
  File "C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\tensorflow_core\python\util\deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\tensorflow_core\python\framework\ops.py", line 3357, in create_op
    attrs, op_def, compute_device)
  File "C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\tensorflow_core\python\framework\ops.py", line 3426, in _create_op_internal
    op_def=op_def)
  File "C:\Users\Joe\Anaconda3\envs\animne\lib\site-packages\tensorflow_core\python\framework\ops.py", line 1748, in __init__
    self._traceback = tf_stack.extract_stack()

**Comment by viet1pro** [comment]
Can you check that there is any program using your GPU and close it. It might be not enough GPU memory

**Comment by JosiMou** [comment]
> Can you check that there is any program using your GPU and close it. It might be not enough GPU memory

managed to fix the cudnn error by putting the follwing code at the very beginning of ai.py

from tensorflow.compat.v1 import ConfigProto
from tensorflow.compat.v1 import InteractiveSession

config = ConfigProto()
config.gpu_options.allow_growth = True
session = InteractiveSession(config=config)
import tensorflow as tf

**Comment by viet1pro** [comment]
This cause becasue your GPU don't have enough memory.
So run these code above will using your CPU with PC Ram(not GPU Ram).
It can make code runable but you will get slower speed.

**Comment by JosiMou** [comment]
It works perfectly!! Thank you so much!
It is super quick actually (got 32gb ram hehe)

**Comment by viet1pro** [comment]
It work quick depend on your CPU :)))

**Comment by viet1pro** [comment]
If it done can you close issue?

**Comment by SNSLogty** [comment]
If you are using GPU to run it,Be sure your GPU has 6G ram or more.

**Comment by SNSLogty** [comment]
We recommand nvidia geforce 1060/1660/2060 or better

**Comment by JosiMou** [comment]
> If it done can you close issue?

yes - thank you so much!

> If you are using GPU to run it,Be sure your GPU has 6G ram or more.

Ahh ok thank you
